packages/lisk-transactions/test/transactions/base.ts

```diff
@@ -17,647 +17,498 @@ import { BaseTransaction } from '../../src/transactions/base';
-						(invalidTransaction as unknown) as TransactionJSON,
-					);
-				} catch (error) {
-					return expect(error).to.be.an.instanceOf(TransactionMultiError);
-				}
-
-				return Promise.resolve();
-			});
+			try {
+				new TestTransaction((invalidTransaction as unknown) as TransactionJSON);
+			} catch (error) {
+				expect(error).to.be.an.instanceOf(TransactionMultiError);
+			}
 		});
 	});
 
-	describe('#assetToJSON', () => {
+	describe('#assetToJSON', async () => {
 		it('should return an object of type transaction asset', () => {
-			return expect(baseTransaction.assetToJSON()).to.be.an('object');
+			expect(baseTransaction.assetToJSON()).to.be.an('object');
 		});
 	});
 
 	describe('#toJSON', () => {
-		it('should return transaction json', () => {
-			baseTransaction = new TestTransaction(defaultTransaction);
+		it('should return transaction json', async () => {
 			const transactionJSON = baseTransaction.toJSON();
 
-			return expect(transactionJSON).to.be.eql(defaultTransaction);
+			expect(transactionJSON).to.be.eql(defaultTransaction);
 		});
 	});
 
 	describe('#getBytes', () => {
-		it('should return a buffer', () => {
+		it('should return a buffer', async () => {
 			expect(baseTransaction.getBytes()).to.be.an.instanceOf(Buffer);
 		});
 	});
 
 	describe('#containsUniqueData', () => {
-		it('should return a boolean', () => {
+		it('should return a boolean', async () => {
 			expect(baseTransaction.containsUniqueData()).to.be.a('boolean');
 		});
 	});
 
 	describe('#checkSchema', () => {
-		describe('when given valid transaction', () => {
-			beforeEach(() => {
-				baseTransaction = new TestTransaction(defaultTransaction);
+		it('should return a successful transaction response with a valid transaction', async () => {
+			const { id, status, errors } = baseTransaction.checkSchema();
 
-				return Promise.resolve();
-			});
+			expect(id).to.be.eql(baseTransaction.id);
+			expect(errors).to.be.eql([]);
+			expect(status).to.eql(Status.OK);
+		});
 
-			it('should return a successful transaction response', () => {
-				const { id, status, errors } = baseTransaction.checkSchema();
+		it('should return a failed transaction response with invalid formatting', async () => {
+			const invalidTransaction = {
+				type: 0,
+				amount: '00001',
+				fee: '0000',
+				recipientId: '',
+				senderPublicKey: '11111111',
+				senderId: '11111111',
+				timestamp: 79289378,
+				asset: {},
+				signature: '1111111111',
+				id: '1',
+			};
+			baseTransaction = new TestTransaction(invalidTransaction as any);
+			const { id, status, errors } = baseTransaction.checkSchema();
 
-				expect(id).to.be.eql(baseTransaction.id);
-				expect(errors).to.be.eql([]);
-				return expect(status).to.eql(Status.OK);
-			});
+			expect(id).to.be.eql(baseTransaction.id);
+			(errors as ReadonlyArray<TransactionError>).forEach(error =>
+				expect(error).to.be.instanceof(TransactionError),
+			);
+			expect(status).to.eql(Status.FAIL);
```

Comment:
Root level beforeEach defines a baseTransaction which we are not using in these tests, maybe it's better to move the baseTransaction creation in the `#constructor` describe block?

Reply:
I will create new constants for invalid transaction tests, but will keep root level baseTransaction as it is used again in later parts of the test as well.